project/SparkBuild.scala

```diff
@@ -443,6 +443,7 @@ object Hive {
         val relGFilePath = (gFilePath relativeTo sourceDir).get.getPath
         log.info("ANTLR: Grammar file '%s' detected.".format(relGFilePath))
         antlr.addGrammarFile(relGFilePath)
+        antlr.setLibDirectory(gFilePath.getParent)
```

Comment:
Tiny thing: By doing this we set the library path multiple times to the same directory. There is unfortunately not an easy way out of this. The solution could be to fix the grammar dir, but that would make the build brittle.

Could you add a comment stating that we assume there is only one library directory?